lib/pymedphys/_experimental/streamlit/apps/transfer_check.py

```diff
@@ -51,17 +67,21 @@ def main():
         else:
             continue
 
+    # If an RP was selected, get plan information from both systems
     if "rp" in files:
-        # retrieve data from both systems.
+
+        # Create a dataframe of plan information from DICOM RP file
         dicom_table = get_all_dicom_treatment_info(files["rp"])
         dicom_table["tolerance"] = [
             TOLERANCE_TYPES[item] for item in dicom_table["tolerance"]
         ]
         dicom_table = dicom_table.sort_values(["field_label"])
 
+        # Using MRN from RP file, find patient in MOSAIQ and perform query
         mrn = dicom_table.iloc[0]["mrn"]
         with connect.connect(server) as cursor:
```

Comment:
Be aware, that connecting to mosaiq like this within a streamlit app will cause it to keep connecting and disconnecting to the server. Streamlit has ways of caching a database connection so that it can stay open. Also, if the user needs to provide a username and password to the database it'll appear in the terminal.

I have made a streamlit mosaiq login function over at:

https://github.com/pymedphys/pymedphys/blob/553732eb8c2f722f7123a66c4e57755db1510881/lib/pymedphys/_streamlit/utilities/mosaiq.py#L20-L38

I'd recommend using the following cached wrapper function:

https://github.com/pymedphys/pymedphys/blob/553732eb8c2f722f7123a66c4e57755db1510881/lib/pymedphys/_streamlit/utilities/mosaiq.py#L88-L90

Reply:
I believe I've done this correctly now... my code still runs which is a good sign. Might need you to double check this one to be sure though!

Reply:
> Might need you to double check this one to be sure though!

Yup, in the next PR I'll have a look :).